File: Solucion de ecuaciones no lineales/newton_raphson.py
# -------------------------------------------------------------------------
# Solucion de ecuaciones no lineales mediante el metodo de Newton-Raphson
# Autor: Alvaro Siesquen
# -------------------------------------------------------------------------
f = lambda x: x**3-155
d = lambda x: 3*x**2

x0 = float(input('Punto inicial: '))
epsilon = float(input('Digite el error: '))
i=1
x = x0 - f(x0)/d(x0)
while(abs(x-x0)>epsilon):
    x0 = x
    x = x0 - f(x0)/d(x0)
    i+=1
print('Raiz calculada: ',x)
print('Numero de iteraciones: ',i)

File: Solucion de ecuaciones no lineales/newton_raphson.f90
!------------------------------------------------------------------------
!Solucion de ecuaciones no lineales mediante el metodo de Newton-Raphson
!Autor: Alvaro Siesquen
!------------------------------------------------------------------------
program newton_raphson
implicit none
!Declaracion de variables
integer i					!Contador
real x0, x, epsilon		!Variables del metodo
real f, d				!Funciones

!Procedimiento
print*,'Punto inicial'
read*,x0
print*,'Digite el error'
read*,epsilon

i=1
x =	x0 - f(x0)/d(x0)

do while (abs(x-x0)>epsilon)
	x0 = x
	x =	x0 - f(x0)/d(x0)
	i = i+1
end do

print*,'Raiz calculada:' ,x
print*,'Numero de iteraciones:' ,i
end program

!Definiendo f(x) y f'(x):
real function f(x)
real x
	f = x**3 - 155
end function

real function d(x)
real x
	d = 3*x**2
end function